blog/2025-07-21-openhab-5-0-release.md

```diff
@@ -0,0 +1,386 @@
+- Basic UI now avoids shrinking the value of a `Text` element and instead shrinks the label.
+
+## Add-on Enhancements
+
+openHAB 5.0 brings xx new add-ons, adding support for new devices:
+
+- TODO
+
+But not only were new bindings added, some existing bindings also received massive improvements:
+
+- TODO
+
+> Probably shout out our top reviewers.
+
+### Z-Wave JS Binding
+
+_Leo Siepel ([@lsiepel](https://github.com/lsiepel)), openHAB Maintainer_
+
+The openHAB 5.0 release introduces a new [Z-Wave JS binding](/addons/bindings/zwavejs), offering an alternative way to integrate Z-Wave networks into openHAB by connecting to an external [Z-Wave JS](https://zwave-js.github.io/zwave-js/) server.
+This binding is particularly useful for advanced users, large networks, or those seeking cross-platform compatibility—such as when migrating from or running alongside other home automation platforms.
+It supports newer 700/800 series chipsets, S2 security, Smart Start, and provides more robust device management features.
+While the binding does not yet expose every capability of [Z-Wave JS](https://zwave-js.github.io/zwave-js/) (such as NVM backups), these can be accessed via the [Z-Wave JS UI](https://zwave-js.github.io/zwave-js-ui/) directly.
+Importantly, the existing native Z-Wave binding remains fully supported, giving users the flexibility to choose the integration that best fits their needs.
+
+More information and discussion can be found on the community forum:
+
+- [New Z-Wave JS binding released](https://community.openhab.org/t/new-z-wave-js-binding-released/164943)
+- [Z-Wave JS UI in place of openHAB Z-Wave binding](https://community.openhab.org/t/zwave-js-ui-in-place-of-oh-zwave-binding/150007)
+
+### MQTT Home Assistant Binding
+
+_Cody Cutrer ([@ccutrer](https://github.com/ccutrer)), openHAB Maintainer_
+
+On the subject of Python, the GraalVM work has enabled the Home Assistant MQTT binding to switch from using the Java Jinjava library for emulating Python's Jinja to actually running Python and Jinja.
+This allows openHAB to import some of the actual Home Assistant Python code, so template rendering from devices should be near-perfect, without constantly playing a game of catch-up as we find devices that use some feature of Python, Jinja, or Home Assistant that we hadn't previously conceived of.
+
+We've also taken this one step further, and re-implemented parsing of the Home Assistant JSON discovery metadata via Home Assistant's own classes, which fixes myriad strange bugs and incompatibilities from attempting to implement in Java using only Home Assistant documentation as a guide.
+We now get the benefit of exactly matching many of those minor idiosyncrasies that aren't documented.
+There's still work to do in the portion of the binding that connects incoming data to openHAB channels.
+However, it's now sitting on a much stronger base so that any issues there can be more quickly iterated on and fixed.
+
```

Comment:
@ccutrer I was just wondering if we could tell people that they can uninstall jinja transform unless they specifically need it outside homeassistant?

Reply:
Wasn't that in 4.3?

Reply:
Ah yes, good memory! I've just re-read the 4.3 blog and it was there! :)